### PR TITLE
fault package funcs rename

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

Renaming the funcs in the fault
package to be more clear about
their purpose and behavior.  Largely
just find&replace changes, except
for fault.go and the fault examples.

## Does this PR need a docs update or release note?

- [x] :no_entry: No 

## Type of change

- [x] :broom: Tech Debt/Cleanup

## Issue(s)

* #1970

## Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Comment by ryanfkeepers** [comment]
@ashmrtn this is ready for review

File: src/internal/operations/manifests.go [hunk 1]
```diff
@@ -163,7 +163,7 @@ func verifyDistinctBases(ctx context.Context, mans []*kopia.ManifestEntry, errs
 			if b, ok := reasons[reasonKey]; ok {
 				failed = true
 
-				et.Add(clues.New("manifests have overlapping reasons").
+				el.AddRecoverable(clues.New("manifests have overlapping reasons").
```
Comment (ashmrtn) [inline src/internal/operations/manifests.go:166]: this should really be a failure as it's not clear what base we should be sourcing items from

File: src/pkg/fault/example_fault_test.go [hunk 2]
```diff
@@ -47,44 +47,40 @@ var dependency = mockDepenedency{}
...
 		return errs
 	}
-	if errs := topLevelHandler(errs); errs.Err() != nil {
-		fmt.Println(errs.Err())
+	if errs := topLevelHandler(errs); errs.Failure() != nil {
+		fmt.Println(errs.Failure())
 	}
 
-	// Only the topmost func in the stack should set the Fail() err.
+	// Only the topmost func in the stack should set the failure.
```
Comment (ashmrtn) [inline src/pkg/fault/example_fault_test.go:83]:
```suggestion
	// Only the top-most func in the stack should set the failure.
```

File: src/pkg/fault/example_fault_test.go [hunk 3]
```diff
@@ -126,154 +130,158 @@ func ExampleErrors_Add() {
...
 			}
 		}
 
-		return trkr.Err()
+		return el.Failure()
 	}()
 	if err != nil {
-		// handle the Err() that appeared in the tracker
-		fmt.Println("err occurred", errs.Err())
+		// handle the Faiulre() that appeared in the local bus.
```
Comment (ashmrtn) [inline src/pkg/fault/example_fault_test.go:262]:
```suggestion
		// handle the Failure() that appeared in the local bus.
```

File: src/pkg/fault/example_fault_test.go [hunk 4]
```diff
@@ -311,36 +320,48 @@ func Example_errors_e2e() {
...
 			if err := storer(d); err != nil {
 				// Since we're at the top of the iteration, we need
-				// to add each error to the fault.Errors struct.
-				errs.Add(err)
+				// to add each error to the fault.localBus struct.
+				el.AddRecoverable(err)
 			}
 		}
+
+		// at the end of the func, we need to return trkr.Failure()
```
Comment (ashmrtn) [inline src/pkg/fault/example_fault_test.go:337]:
```suggestion
		// at the end of the func, we need to return local.Failure()
```

File: src/pkg/backup/backup.go [hunk 5]
```diff
@@ -37,7 +37,7 @@ type Backup struct {
 	Version int `json:"version"`
 
 	// Errors contains all errors aggregated during a backup operation.
-	Errors fault.ErrorsData `json:"errors"`
+	Errors fault.Errors `json:"errors"`
```
Comment (ashmrtn) [inline src/pkg/backup/backup.go:40]: this needs tested (manually or unit test) to ensure it still allows deserializing backups that had errors

**Review by ashmrtn — APPROVED** [review]
needs some testing for serialization before merging

File: src/internal/operations/manifests.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/operations/manifests.go:166]: Done

File: src/pkg/fault/fault.go [hunk 6]
```diff
@@ -6,21 +6,20 @@ import (
...
 )
 
-type Errors struct {
+type Bus struct {
 	mu *sync.Mutex
 
-	// err identifies non-recoverable errors.  This includes
+	// failure identifies non-recoverable errors.  This includes
 	// non-start cases (ex: cannot connect to client), hard-
 	// stop issues (ex: credentials expired) or conscious exit
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/pkg/fault/fault.go]: Shaped up the language a bit.

File: src/pkg/backup/backup.go (hunk 5 above)
Comment (ryanfkeepers) [reply src/pkg/backup/backup.go:40]: confirmed all green

**Pull request merged**